mcu/attiny-hal/src/adc.rs

```diff
@@ -68,6 +68,7 @@ pub mod channel {
     pub struct Temperature;
 }
 
+#[cfg(not(feature = "attiny84a"))]
```

Comment:
So, this is a multi-level issue:

First of all, to not build the ADC module for this chip, it would be better to disable it at crate level.  Here:

https://github.com/Rahix/avr-hal/blob/21342dcace7184f01fdc4e9703b01197bd4b4b4f/mcu/attiny-hal/src/lib.rs#L65-L69

However, the chip does have an ADC so this code shouldn't fail to compile.  The reason it does is probably that registers in `avr-device` are not correctly defined.  I've already investigated this and opened https://github.com/Rahix/avr-device/pull/151 to fix it.

So I think the best course of action is to wait for the change to land in `avr-device` and then things should just work without the `#[cfg(not())]` line.